src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs

```diff
@@ -92,7 +92,23 @@ private ClientConfigPaths(string exePath, bool includeUserConfig)
                 }
             }
 
-            if (!string.IsNullOrEmpty(ApplicationUri))
+            string externalConfigPath = AppDomain.CurrentDomain.GetData("APP_CONFIG_FILE") as string;
+            if (!string.IsNullOrEmpty(externalConfigPath))
+            {
+                if (Uri.IsWellFormedUriString(externalConfigPath, UriKind.Absolute))
+                {
+                    Uri externalConfigUri = new Uri(externalConfigPath);
+                    if (externalConfigUri.IsFile)
+                    {
+                        ApplicationConfigUri = externalConfigUri.LocalPath;
+                    }
+                }
+                else
+                {
+                    ApplicationConfigUri = Path.GetFullPath(externalConfigPath);
```

Comment:
This is implicitly prefixing with whatever the current directory is at the time the config system is initialized, I guess?

In the desktop code it is using the appbase to prefix (naming.cpp line 3241 .. assuming I'm looking in the right place)

I think in .NET Core that is AppDomain.BaseDirectory, so should this be `Path.Combine(AppDomain.CurrentDomain.BaseDirectory, externalConfigPath)` ? There is something similar higher up.

Reply:
You can change the tests to set Environment.CurrentDirectory to something random before reading the value (I'm guessing the test will fail right now)

Reply:
When you use `Path.Combine` you need to be careful with paths that begin with `\`, as you'll get the second path against the root of the current directory. Trim the start of the `externalConfigPath` for safety.

Reply:
I have used `Path.GetFullPath(externalConfigPath, AppDomain.CurrentDomain.BaseDirectory)` which seems to be working fine. The tests are now changing CurrentDirectory to validate we are relative to the BaseDirectory and not CurrentDirectory

Reply:
hmm seems that overload is not available everywhere...